library/java/src/io/envoyproxy/envoymobile/engine/JvmFilterContext.java

```diff
@@ -0,0 +1,104 @@
+package io.envoyproxy.envoymobile.engine;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+import io.envoyproxy.envoymobile.engine.types.EnvoyHTTPFilter;
+
+/**
+ * Wrapper class for EnvoyHTTPFilter for receiving JNI calls.
+ */
+class JvmFilterContext {
```

Comment:
Is it possible for us to add unit tests to just ensure the the filters are called appropriately?

Reply:
https://github.com/lyft/envoy-mobile/pull/1024

Reply:
Could we also have a `JvmFilterContext` unit test which helps us with exercising the interaction between the `JvmBridgeUtility` and `EnvoyHTTPFilter`?

Reply:
Stepping back for a second, I'm honestly skeptical of the value of **unit** testing functions that purely perform delegation. There's no real black box logic to test; the function makes another call, and you implement that call on one hand and then assert you've implemented it on the other. In my opinion, integration testing is a much better mechanism for testing for correct wiring/coupling of layers.

Translation on the other hand, I think merits unit testing - which is partly why I've extracted logic into the `JvmBridgeUtility` (and also the `JniBridgeUtility` in the upcoming PR).

Reply:
That's fair. I would like to capture and exercise the delegation path somewhere.

I assume you have in mind an integration test which would exercise this?

Reply:
Yes, I'd like to cover the paths through the layer wiring using the AssertionFilter that @junr03 has implemented.